__tests__/e2e/utils/mail.ts

```diff
@@ -132,3 +123,31 @@ const getSubmissionAttachments = async (
   }
   return atts
 }
+
+/**
+ * Retrieves an autoreply email sent by FormSG.
+ * @param {string} responseId response ID of the submission
+ * @param {string} recipient email address of the form filler
+ * @returns {MailData} email for the autoreply sent to the recipient
+ */
+export const getAutoreplyEmail = async (
+  responseId: string,
+  recipient: string,
+): Promise<MailData> => {
+  const emails = await getEmailsBy(
+    (email) =>
+      email.to[0].address === recipient &&
+      email.html.includes(`Response ID: ${responseId}`),
+  )
+
+  const lastEmail = emails.pop()
+  if (!lastEmail) {
+    throw Error(
+      `mailbox does not contain autoreply email for response ID ${responseId}`,
+    )
+  }
+
+  await MAIL_CLIENT.deleteById(lastEmail.id)
```

Comment:
Haha this bug plagued me for a solid half a day :') 

There may be interference between OTP field verification emails and autoreply emails, since they're both sent to the same email address (i.e. the one that was entered in the email field). The way the tests are written, the OTP email getter simply pops the latest email from the inbox which was received by the recipient. If multiple tests use the same email field value, then an autoreply email may be picked up rather than an OTP email. 

It might have been possible to put a better filter in the extractOtp function to get emails with a particular subject, but it would also have been hacky because the email subject for admin logins and OTP verifications isn't the same. We also can't filter by regex because response IDs are in the body of autoreply emails, and may have 6 consecutive digits within their body. Anyway this seemed to work fine for the couple test runs I did, so I figured we can leave it as is for now. If it becomes flaky later on, we can look at improving it!